src/Symfony/Component/ImportMaps/ImportMapManager.php

```diff
@@ -0,0 +1,311 @@
+            if (isset($data['url'])) {
+                $importmap['imports'][$packageName] = ($data['download'] ?? false) ? $this->vendorUrl($packageName) : $data['url'];
+            } elseif (isset($data['path'])) {
+                $importmap['imports'][$packageName] = $this->assetsUrl.$this->digestName($packageName, $data['path']);
+            } else {
+                continue;
+            }
+
+            if ($data['preload'] ?? false) {
+                $this->modulesToPreload[] = $importmap['imports'][$packageName];
+            }
+        }
+
+        // Use JSON_UNESCAPED_SLASHES | JSON_HEX_TAG to prevent XSS
+        $this->json = json_encode($importmap, \JSON_THROW_ON_ERROR | \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES | \JSON_HEX_TAG);
+    }
+
+    /**
+     * @param array<string, PackageOptions> $require
+     * @param string[]                      $remove
+     */
+    private function createImportMap(bool $update, array $require, array $remove): void
+    {
+        $this->loadImportMap();
+
+        foreach ($remove as $packageName) {
+            if (!isset($this->importMap[$packageName])) {
+                continue;
+            }
+
+            $this->cleanup($this->importMap, $packageName);
+            unset($this->importMap[$packageName]);
+        }
+
+        $install = [];
+        $packages = [];
+        foreach ($this->importMap ?? [] as $packageName => $data) {
+            if (isset($data['path'])) {
+                $publicPath = $this->publicAssetsDir.$this->digestName($packageName, $data['path']);
+                if (file_exists($publicPath)) {
+                    continue;
+                }
+
+                $this->cleanup($this->importMap, $packageName, false);
+                @mkdir($this->publicAssetsDir, 0777, true);
+                copy($this->assetsDir.$data['path'], $publicPath);
+
+                continue;
+            }
+
+            $packages[$packageName] = new PackageOptions($data['download'] ?? false, $data['preload'] ?? false);
+            if (preg_match(self::PACKAGE_PATTERN, $data['url'] ?? $packageName, $matches)) {
+                $constraint = ($matches['registry'] ?? null) ? "{$matches['registry']}:{$matches['package']}" : $matches['package'];
+
+                if (!$update && ($matches['version'] ?? null)) {
+                    $constraint .= "@{$matches['version']}";
+                }
+
+                $install[$matches['package']] = $constraint;
+            }
+        }
+
+        foreach ($require as $packageName => $packageOptions) {
+            if (preg_match(self::PACKAGE_PATTERN, $packageName, $matches)) {
+                $install[$matches['package']] = $packageName;
+                $packages[$matches['package']] = $packageOptions;
+            }
+        }
+
+        $this->jspmGenerate($install, $packages);
+
+        file_put_contents(
+            $this->path,
+            sprintf("<?php\n\nreturn %s;\n", VarExporter::export($this->importMap)),
```

Comment:
let's use var_export if VarExporter is not installed to make the dep to var-exporter optional